library/src/main/scala/scala/scalajs/js/BigInt.scala

```diff
@@ -0,0 +1,116 @@
+ *
+ * Licensed under Apache License 2.0
+ * (https://www.apache.org/licenses/LICENSE-2.0).
+ *
+ * See the NOTICE file distributed with this work for
+ * additional information regarding copyright ownership.
+ */
+
+package scala.scalajs.js
+
+import scala.scalajs.js
+import scala.scalajs.js.annotation.JSGlobal
+
+/** <span class="badge badge-ecma2019" style="float: right;">ECMAScript 2019</span>
+ *
+ *  A built-in object that provides a way to represent whole numbers larger than
+ *  2 ^ 53 - 1, which is the largest number JavaScript can reliably represent
+ *  with the Number primitive.
+ *
+ *  BigInt can be used for arbitrarily large integers.
+ *
+ *  @see https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/BigInt
+ */
+@js.native
+@JSGlobal
+final class BigInt private[this] () extends js.Object {
+
+  def +(other: BigInt): BigInt = js.native
+  def *(other: BigInt): BigInt = js.native
+  def -(other: BigInt): BigInt = js.native
+  def **(other: BigInt): BigInt = js.native
+  def %(other: BigInt): BigInt = js.native
+
+  def &(other: BigInt): BigInt = js.native
+  def |(other: BigInt): BigInt = js.native
+  def ^(other: BigInt): BigInt = js.native
+  def <<(other: BigInt): BigInt = js.native
+  def >>(other: BigInt): BigInt = js.native
+  // no >>> since BigInt is always signed
+
+  // scalastyle:off disallow.space.before.token
+  def unary_- : BigInt = js.native
+  def unary_~ : BigInt = js.native
+  // no unary_+ not to break asm.js
+  // scalastyle:on disallow.space.before.token
+
+  def <(x: BigInt): Boolean = js.native
+  def <=(x: BigInt): Boolean = js.native
+  def >(x: BigInt): Boolean = js.native
+  def >=(x: BigInt): Boolean = js.native
+
+  /**
+   *
+   * @see https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/BigInt/toLocaleString
+   * @param locale A string with a BCP 47 language tag, or an array of such
+   *               strings. For the general form and interpretation of the
+   *               locales argument, see the Intl page.
+   * @param options An object with some or all of the properties.
+   * @return
+   */
+  def toLocaleString(locale: String, options:
+      BigInt.ToLocaleStringOptions = js.native): String = js.native
+
+  /**
+   * @see https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/BigInt/toString
+   * @return A string representing the specified BigInt object.
+   *         The trailing "n" is not part of the string.
+   */
+  override def toString(): String = js.native
+}
+
+
+/** <span class="badge badge-ecma2019" style="float: right;">ECMAScript 2019</span>
+ *
+ *  A companion object of BigInt class.
+ *
+ *  @see https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/BigInt
+ */
+@js.native
+@JSGlobal
+object BigInt extends js.Object {
+  /** @see https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/BigInt */
+  def apply(value: Int): BigInt = js.native
```

Comment:
I think we should also expose one more overload that takes a `Double`. It won't be completely "safe", since fractional numbers will cause a `RangeError`, but if we *don't* expose it, then there is no way to create a `BigInt` from a whole `Double` that is greater than `Int.MaxValue`.

Reply:
Added `Double`-overload